### ISSUE TITLE
[PW-3330] Google Pay is not available 

### ISSUE THREAD

**Issue by sean-norris** [Adyen/adyen-magento2#869]
**Describe the bug**
When checking out and selecting Google Pay and error message states "Google Pay is not available". This issue has first started occuring about 4-6 weeks ago and we haven't been able to accept a Google Payment since.

**To Reproduce**
1. Add items to cart
2. Click on checkout
3. Select Google Pay as the payment method

**Expected behavior**
Once you choose Google Pay you should be able to select to complete your order

**Screenshots**
![Screenshot 2020-11-03 122205](https://user-images.githubusercontent.com/63900135/97949693-6d0df880-1dcf-11eb-9a50-6e3f2dcb9d97.jpg)

**Additional context**
We were asked to upgrade to version 6.6.1 of the Magento plugin which we did but it hasn't resolved the error. We use Magento 2.3.5

**Comment by msilvagarcia** [comment]
Hey @sean-norris !

Do you see any errors or warnings in the console? Maybe there's some clues there :)

Kind regards,
Marcos
Adyen

**Comment by sean-norris** [comment]
Hi @msilvagarcia 

Yep, good call. There is this error relating to the problem. Any idea on what causes this issue? 
![image](https://user-images.githubusercontent.com/63900135/97990444-ea128f80-1e1a-11eb-9244-5c7107fefdef.png)

The other payment methods work fine and Google Pay did up until recently.

**Comment by AlexandrosMor** [comment]
Hello @sean-norris,

It seems that the plugin is not installed properly. Could you maybe re install and test with the latest version ?  

kind regards,
Alexandros
Adyen

**Comment by sean-norris** [comment]
Hi @AlexandrosMor 

We have upgraded to the latest available plugin and still facing the same issue. We are investigating further from our side but if you can offer any further assistance it would be appreciated.

**Comment by sean-norris** [comment]
Hi @AlexandrosMor 

We have upgraded to the latest available plugin and still facing the same issue. We are investigating further from our side but if you can offer any further assistance it would be appreciated.

**Comment by cyattilakiss** [comment]
Hi @sean-norris ,

Just to be sure are you testing it with a google pay compatible device/browser while you are signed in with a user which has a payment method already set up in their wallet?

If so, could you please check if these configurations have the correct value?
https://github.com/Adyen/adyen-magento2/blob/6.6.1/view/frontend/web/js/view/payment/method-renderer/adyen-google-pay-method.js#L107-L111

If yes, do you see if the promise here: https://github.com/Adyen/adyen-magento2/blob/6.6.1/view/frontend/web/js/view/payment/method-renderer/adyen-google-pay-method.js#L144-L151
mounts the component: https://github.com/Adyen/adyen-magento2/blob/6.6.1/view/frontend/web/js/view/payment/method-renderer/adyen-google-pay-method.js#L146 or logs an error: https://github.com/Adyen/adyen-magento2/blob/6.6.1/view/frontend/web/js/view/payment/method-renderer/adyen-google-pay-method.js#L149 ?

Best,
Attila
Adyen

**Comment by atty31** [comment]
Hi, 
For me the following cases are triggered:

- On firefox(version 82.0.3 ), I'm receiving the following js error:
```
Uncaught ReferenceError: PaymentRequest is not defined
    <anonymous> https://pay.google.com/gp/p/js/pay.js:240
```

- On chrome (version: 86.0.4240.198 ) I got a report, saying:
`adyen-methods.js:73 [Report Only] Refused to load the script 'https://pay.google.com/gp/p/js/pay.js' because it violates the following Content Security Policy directive`

Also,I cannot click on the following button:
![Uploading Screenshot(25).png…]()

I'm running the adyen module version 6.6.5 on Magento 2.4. Any thoughts what is going on ?  Many thanks!

**Comment by cyattilakiss** [comment]
Hi @atty31 ,

I see that the google pay button is rendered for you correctly so your configurations are correct.

Please enable Terms and Conditions for the checkout in Magento and then after the customer accepts the T&C, the google pay button will be enabled. We are meanwhile working on removing this "dependency" so merchants without T&C checkbox should be able to use it after that fix as well.

Enable T&C steps:
1. Login to admin panel.
1. Go to Stores > Configuration.
1. Under Sales, select Checkout.
1. Expand the Checkout Options.
1. Set “Enable Terms and Conditions” field to YES.
1. Click Save Config.

Best,
Attila
Adyen

**Comment by atty31** [comment]
Hi @cyattilakiss ,

Thnx for the clear explanation (Köszi szépen :+1:  ). The only remark I would like to add is that this is working if the T&C are set to **manual**. See img: https://imgur.com/mab47KH . IF it is set to automatically the button is still disabled 

Thank you and have a great day !

**Comment by cyattilakiss** [comment]
Hi @atty31 ,

Good feedback, thanks!
We are picking up this issue in the next sprint starting on Monday.

Best,
Attila
Adyen

**Comment by floorjubbega** [comment]
Hi @atty31 and @sean-norris, 

This issue has been fixed in release 6.6.6.
We suggest to update to our latest version, and let us know if you see any further issues with regards to Google Pay. 

Thank you!

Regards,
Floor 
Adyen
